include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -191,6 +192,328 @@ AffineConstraints<number>::is_consistent_in_parallel(
+
+    // step 1: identify owners of constraints
+    std::vector<unsigned int> constrained_indices_owners(
+      constrained_indices.n_elements());
+    Utilities::MPI::internal::ComputeIndexOwner::ConsensusAlgorithmsPayload
+      constrained_indices_process(locally_owned_dofs,
+                                  constrained_indices,
+                                  mpi_communicator,
+                                  constrained_indices_owners,
+                                  true);
+
+    Utilities::MPI::ConsensusAlgorithms::Selector<
+      std::pair<types::global_dof_index, types::global_dof_index>,
+      unsigned int>(constrained_indices_process, mpi_communicator)
+      .run();
+
+    // step 2: collect all locally owned constraints
+    const auto constrained_indices_by_ranks =
+      constrained_indices_process.get_requesters();
+
+    {
+      const unsigned int tag = Utilities::MPI::internal::Tags::
+        affine_constraints_make_consistent_in_parallel_0;
+
+      // ... collect data and sort according to owner
+      std::map<unsigned int, std::vector<ConstraintType>> send_data_temp;
+
+      for (unsigned int i = 0; i < constrained_indices_owners.size(); ++i)
+        {
+          ConstraintType entry;
+
+          const types::global_dof_index index =
+            constrained_indices.nth_index_in_set(i);
+
+          std::get<0>(entry) = index;
+
+          if (is_inhomogeneously_constrained(index))
+            std::get<1>(entry) = get_inhomogeneity(index);
+
+          const auto constraints = get_constraint_entries(index);
+
+          if (constraints)
+            for (const auto &i : *constraints)
+              std::get<2>(entry).push_back(i);
+
+          if (constrained_indices_owners[i] == my_rank)
+            locally_relevant_constrains.push_back(entry);
+          else
+            send_data_temp[constrained_indices_owners[i]].push_back(entry);
+        }
+
+      std::map<unsigned int, std::vector<char>> send_data;
+
+      for (const auto &i : send_data_temp)
+        send_data[i.first] = Utilities::pack(i.second, false);
+
+      std::vector<MPI_Request> requests;
+      requests.reserve(send_data.size());
+
+      // ... send data
+      for (const auto &i : send_data)
+        {
+          if (i.first == my_rank)
+            continue;
+
+          requests.push_back({});
+
+          const int ierr = MPI_Isend(i.second.data(),
+                                     i.second.size(),
+                                     MPI_CHAR,
+                                     i.first,
+                                     tag,
+                                     mpi_communicator,
+                                     &requests.back());
+          AssertThrowMPI(ierr);
+        }
+
+      // ... receive data
+      std::set<unsigned int> rec_ranks;
+
+      for (const auto &i : constrained_indices_by_ranks)
+        if (i.first != my_rank)
+          rec_ranks.insert(i.first);
```

Comment:
Why do you need `rec_ranks`? It seems you only need `constrained_indices_by_ranks.size()-1` for the loop below.

Reply:
I need to guarantee that my_rank is skipped.

Reply:
But even then you don't need to fill a set, all you need is count the appearances, don't you?

Reply:
You are right.